Fork_Maintenance.md

```diff
@@ -0,0 +1,263 @@
+# OpenHands Fork Maintenance Guide
+
+This guide outlines a strategy for maintaining a fork of the main OpenHands repository while incorporating custom patches and ensuring stability and security.
+
+## Key Goals
+
+1.  **Consistent Container Images:** Use a specific, matched pair of `runtime` and `agent` images across all environments.
+2.  **Security & Intentional Updates:** Avoid automatic updates from upstream; intentionally review and integrate changes.
+3.  **Local Patch Management:** Easily add, manage, and remove custom features or external PRs.
+
+## 1. Initial Setup
+
+First, ensure your fork is cloned locally and the original OpenHands repository is added as a remote named `upstream`.
+
+```bash
+# Clone your fork (replace with your fork URL)
+# git clone <your-fork-url>
+# cd OpenHands
+
+# Add the main OpenHands repository as the 'upstream' remote
+git remote add upstream https://github.com/All-Hands-AI/OpenHands.git
+
+# Fetch upstream branches and tags
+git fetch upstream
+```
+
+## 2. Container Image Management
+
+To ensure consistency, build, tag, and distribute your own `runtime` and `agent` images based on your primary integration branch.
+
+**a. Building Images:**
+
+*   **Source Branch:** Images should be built by CI/CD processes triggered by commits to the `release/stable-with-patches` branch.
+*   **Build Process:** Use the standard OpenHands build process (`docker build`, `make`, etc.).
+*   **Base Image Consideration:** While the image is built *from* the code in `release/stable-with-patches`, consider how the underlying base OS/dependency image specified in the Dockerfile is kept up-to-date. This might involve periodically updating the base image tag in the Dockerfile itself as part of the fork maintenance.
+
+*(Example - Adapt based on actual build commands, assuming build context is repo root)*
+```bash
+# Example: Build triggered by CI on release/stable-with-patches
+git checkout release/stable-with-patches
+MAIN_COMMIT_BASE=$(git log -n 1 --pretty=%H main) # Get the latest main commit integrated
+BUILD_TAG="stable-$(date +%Y%m%d)-$(git rev-parse --short HEAD)-main-${MAIN_COMMIT_BASE:0:7}"
```

Comment:
Tagging with a date isn't super useful if we want all our other repos to not require changes when we update our images.

In other words, I expect other repos to just use `:latest-stable` whenever they reference openhands. So an update to latest-stable applies everywhere.